yo/db.py

```diff
@@ -405,7 +360,122 @@ def set_user_transports(self, username=None, transports=None):
                         'Exception occurred trying to update transports for user %s to %s' % (
                             username, str(transports)))
                 result = self.create_user(username, transports=transports)
-                logger.debug('set_user_transpors insert result: %s', result)
+                logger.debug('set_user_transports insert result: %s', result)
                 if result:
-                    return transports
-            return False
+                    return  transports
+                return None
+
+
+
+    def get_priority_count(self,
+                           to_username,
+                           priority,
+                           timeframe,
+                           start_time=None):
+        """Returns count of notifications to a user of a set priority or higher
+
+       This is used to implement the rate limits
+
+       Args:
+           to_username(str): The username to lookup
+           priority(int):    The priority level to lookup
+           timeframe(int):   The timeframe in seconds to check
+
+       Keyword args:
+           start_time(datetime.datetime): the current time to go backwards from, if not set datetime.now() will be used
+
+       Returns:
+           An integer count of the number of notifications sent to the specified user within the specified timeframe of that priority level or higher
+           :param timeframe:
+           :param priority:
+           :param to_username:
+           :param start_time:
+       """
+        if start_time is None:
+            start_time = datetime.datetime.now() - datetime.timedelta(
+                seconds=timeframe)
+        retval = 0
+        with self.acquire_conn() as conn:
+            try:
+                query = notifications_table.select().where(
+                    notifications_table.c.to_username == to_username)
+                query = query.where(
+                    notifications_table.c.priority_level >= priority)
+                query = query.where(notifications_table.c.sent == True)
+                query = query.where(
+                    notifications_table.c.sent_at >= start_time)
+                select_response = conn.execute(query)
+                retval = select_response.rowcount
+            except:
+                logger.exception('Exception occurred!')
+        return retval
+
+    def create_wwwpoll_notification(self,notify_id=None,notify_type=None,created_time=None,raw_data={},to_username=None,shown=False,read=False):
```

Comment:
With regard to reverting changes, please be specific - some changes were caught up in the merge, others needed to be changed as they were breaking stuff.

As for raw_data={}, there are scenarios where no raw_data is needed, but everything assumes it to be a dict, hence using an empty dict instead of None.